In [5]:
def divide_byc(a,b,c):
  sum=0
  for i in range(a,b+1):
    if i%c==0:
      sum+=i
  return sum
print(divide_byc(1,10,20))
print(divide_byc(1,10,2))
print(divide_byc(1,10,3))


0
30
18


Question1. Create a function that takes three arguments a, b, c and returns the sum of the
numbers that are evenly divided by c from the range a, b inclusive.
Examples
evenly_divisible(1, 10, 20) ➞ 0
# No number between 1 and 10 can be evenly divided by 20.
evenly_divisible(1, 10, 2) ➞ 30
# 2 + 4 + 6 + 8 + 10 = 30
evenly_divisible(1, 10, 3) ➞ 18
# 3 + 6 + 9 = 18
answer:-
def divide_byc(a,b,c):
  sum=0
  for i in range(a,b+1):
    if i%c==0:
      sum+=i
  return sum
print(divide_byc(1,10,20))
print(divide_byc(1,10,2))
print(divide_byc(1,10,3))

Question2. Create a function that returns True if a given inequality expression is correct and
False otherwise.
Examples
correct_signs(&quot;3 &lt; 7 &lt; 11&quot;) ➞ True
correct_signs(&quot;13 &gt; 44 &gt; 33 &gt; 1&quot;) ➞ False
correct_signs(&quot;1 &lt; 2 &lt; 6 &lt; 9 &gt; 3&quot;) ➞ True
answer:-
def correct_signs(inequality_expression):
  """Returns True if a given inequality expression is correct and False otherwise.

  Args:
    inequality_expression: A string representing the inequality expression.

  Returns:
    A boolean value indicating whether the inequality expression is correct.
  """

  # Split the inequality expression into a list of tokens.
  tokens = inequality_expression.split()

  # Check if the number of tokens is odd.
  if len(tokens) % 2 != 1:
    return False

  # Iterate over the tokens, comparing each pair of adjacent tokens.
  for i in range(0, len(tokens), 2):
    # Get the two adjacent tokens.
    token1 = tokens[i]
    token2 = tokens[i + 1]

    # Check if the two tokens are numbers.
    if not token1.isdigit() or not token2.isdigit():
      return False

    # Convert the two tokens to integers.
    number1 = int(token1)
    number2 = int(token2)

    # Check if the inequality sign is correct.
    if token1 == "<":
      if number1 >= number2:
        return False
    elif token1 == ">":
      if number1 <= number2:
        return False

  # If we reach this point, then all of the inequality signs are correct.
  return True


Question3. Create a function that replaces all the vowels in a string with a specified character.
Examples
replace_vowels(&quot;the aardvark&quot;, &quot;#&quot;) ➞ &quot;th# ##rdv#rk&quot;
replace_vowels(&quot;minnie mouse&quot;, &quot;?&quot;) ➞ &quot;m?nn?? m??s?&quot;
replace_vowels(&quot;shakespeare&quot;, &quot;*&quot;) ➞ &quot;sh*k*sp**r*&quot;
answer:-
def replace_vowels(a,b):
  vowels = ['a','e','i','o','u']
  for i in a:
    if i in vowels:
       c = a.replace(i,b)
  return c

  
replace_vowels('the aardvark', '#')
replace_vowels('minnie mouse', '?')
replace_vowels('shakespeare', '*')

In [26]:
def factorial(a):
  if a==0:
    return 1
  else:
    return a * factorial(a-1)
print(factorial(5))
print(factorial(3))
print(factorial(1))
print(factorial(0))

120
6
1
1


Question4. Write a function that calculates the factorial of a number recursively.
Examples
factorial(5) ➞ 120
factorial(3) ➞ 6
factorial(1) ➞ 1
factorial(0) ➞ 1
answer:-
def factorial(a):
  if a==0:
    return 1
  else:
    return a * factorial(a-1)
print(factorial(5))
print(factorial(3))
print(factorial(1))
print(factorial(0))

In [ ]:
|

Question 5
Hamming distance is the number of characters that differ between two strings.
To illustrate:
String1: &quot;abcbba&quot;
String2: &quot;abcbda&quot;
Hamming Distance: 1 - &quot;b&quot; vs. &quot;d&quot; is the only difference.
Create a function that computes the hamming distance between two strings.
Examples
hamming_distance(&quot;abcde&quot;, &quot;bcdef&quot;) ➞ 5
hamming_distance(&quot;abcde&quot;, &quot;abcde&quot;) ➞ 0
hamming_distance(&quot;strong&quot;, &quot;strung&quot;) ➞ 1
answer:-
def hamming_distance(str1, str2):
  """Computes the Hamming distance between two strings.

  Args:
    str1: A string representing the first string.
    str2: A string representing the second string.

  Returns:
    An integer representing the Hamming distance between the two strings.
  """

  if len(str1) != len(str2):
    return -1

  distance = 0
  for i in range(len(str1)):
    if str1[i] != str2[i]:
      distance += 1

  return distance

